### PR TITLE
Clean up LocalKeyStore and related

### PR DESCRIPTION
**Pull request by Valodim**
I introduced a bug in #3744 by injecting LocalKeyStore via Koin before the storage path is initialized. This PR cleans up LocalKeyStore and related classes, and they are now properly dependency injected.

### CODE REVIEW

File: app/core/src/main/java/com/fsck/k9/LocalKeyStoreManager.kt [hunk 1]
```diff
@@ -7,18 +7,17 @@ import java.security.cert.CertificateException
...
 ) {
     /**
      * Add a new certificate for the incoming or outgoing server to the local key store.
      */
     @Throws(CertificateException::class)
     fun addCertificate(account: Account, direction: MailServerDirection, certificate: X509Certificate) {
-        val uri: Uri
-        if (direction === MailServerDirection.INCOMING) {
-            uri = Uri.parse(account.storeUri)
+        val uri = if (direction === MailServerDirection.INCOMING) {
```
Comment (cketti) [inline app/core/src/main/java/com/fsck/k9/LocalKeyStoreManager.kt]: Side note: Even nicer with the `when` keyword. You can use `MailServerDirection.INCOMING` and `MailServerDirection.OUTGOING` without needing an else case. Builds will automatically fail when another enum value is added without this `when` being changed to add the missing case.

File: mail/common/src/main/java/com/fsck/k9/mail/ssl/LocalKeyStore.java [hunk 2]
```diff
@@ -11,63 +11,37 @@
...
-    }
-
-    private static class LocalKeyStoreHolder {
-        static final LocalKeyStore INSTANCE = new LocalKeyStore();
-    }
-
-    public static LocalKeyStore getInstance() {
-        return LocalKeyStoreHolder.INSTANCE;
+    public static LocalKeyStore createInstance(Context context) {
+        String keyStoreLocation = context.getDir("KeyStore", Context.MODE_PRIVATE).toString();
```
Comment (cketti) [inline mail/common/src/main/java/com/fsck/k9/mail/ssl/LocalKeyStore.java]: This adds a dependency to an Android class. Ideally we want to push these Android dependencies as far "up" as possible.

Comment (Valodim) [reply mail/common/src/main/java/com/fsck/k9/mail/ssl/LocalKeyStore.java]: I considered this but couldn't find an obvious solution that doesn't add a lot of code. Do you have a suggestion on how to do that here? Introduce a new K9Context interface and specialize it for methods like this?

**Comment by Valodim** [comment]
I'll merge this for now because it breaks master for me. Feel free to fix the small dependency regression, suggest a fix, or leave as is.

**Pull request merged**